design/volume-snapshot-data-movement/volume-snapshot-data-movement.md

```diff
@@ -663,6 +662,32 @@ If DM restarts, Velero has no way to detect this, DM is expected to:
 
 At present, VBDM doesn't support recovery, so it will follow the second rule.  
 
+## Kopia For Block Device
+To work with block devices, VGDP will be updated. Today, when Kopia attempts to create a snapshot of the block device, it will error because kopia does not support this file type. Kopia does have a nice set of interfaces that are able to be extended though.
+
+To achieve the necessary information to determine the type of volume that is being used, we will need to pass in the volume mode in provider interface.
+
+```go
+// Provider which is designed for one pod volume to do the backup or restore
+type Provider interface {
+	// RunBackup which will do backup for one specific volume and return snapshotID, isSnapshotEmpty, error
+	// updater is used for updating backup progress which implement by third-party
+	RunBackup(
+		ctx context.Context,
+		path string,
+		volMode v1.PersistentVolumeMode,
+		realSource string,
+		tags map[string]string,
+		forceFull bool,
+		parentSnapshot string,
+		updater uploader.ProgressUpdater) (string, bool, error)
+```
+
+In this case, we will extend the default kopia uploader to add the ability, when a given volume is for a block mode and is mapped as a device, we will use a new specific type that conforms to the [File](https://pkg.go.dev/github.com/kopia/kopia/fs@v0.13.0#File) interface. This new type, will handle streaming the block device to the rest of the kopia process.
+
+While restoring, we are able extend the [Output](https://pkg.go.dev/github.com/kopia/kopia@v0.13.0/snapshot/restore#Output) interface. This will allow us to handle the case where we are writing to the device
+
+While extending these interfaces, we will use the defaults except for when it is for a block device.
```

Comment:
I added the code on how it will extend the interfaces please let me know if this works. For now resolving this convo but we can re-open or start a new one if we need to.